source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1999,59 +2000,139 @@ void COpenGLDriver::SAuxContext::flushState(GL_STATE_BITS stateBits)
     }
 #undef STATE_NEQ
 #undef UPDATE_STATE
-    if ((stateBits & GSB_DESCRIPTOR_SETS) && currentState.pipeline)
+    if (stateBits & GSB_DESCRIPTOR_SETS)
     {
-        //TODO get rid of constants and use runtime-queried binding point counts
-        GLuint ubonames[MAX_UBO_BINDING_COUNT]{};
-        GLintptr ubooffsets[MAX_UBO_BINDING_COUNT]{};
-        GLsizeiptr ubosizes[MAX_UBO_BINDING_COUNT]{};
-
-        GLuint ssbonames[MAX_SSBO_BINDING_COUNT]{};
-        GLintptr ssbooffsets[MAX_SSBO_BINDING_COUNT]{};
-        GLsizeiptr ssbosizes[MAX_SSBO_BINDING_COUNT]{};
-
-        GLuint texnames[MAX_TEXTURE_BINDING_COUNT]{};
-        GLuint smplrnames[MAX_TEXTURE_BINDING_COUNT]{};
-
-        GLuint imgnames[MAX_IMAGE_BINDING_COUNT]{};
-        GLenum imgformats[MAX_IMAGE_BINDING_COUNT]{};
+        //provide storage for multibind calls' parameters
+        constexpr size_t UNIT_SIZE = 5*sizeof(GLuint) + 2*sizeof(GLintptr) + 2*sizeof(GLsizeiptr);
+        uint8_t stackmem[1u<<14]{};//16k bytes
```

Comment:
why do you stil need this storage?

Reply:
see https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r333204559